lib/block-supports/layout.php

```diff
@@ -25,6 +25,108 @@ function gutenberg_register_layout_support( $block_type ) {
 	}
 }
 
+function gutenberg_generate_common_flow_layout_styles() {
+	$style_engine = WP_Style_Engine_Gutenberg::get_instance();
+
+	$style_engine->add_style(
+		'wp-layout-flow',
+		array(
+			'selector' => '.alignleft',
+			'rules'    => array(
+				'float'        => 'left',
+				'margin-right' => '2em',
+				'margin-left'  => '0',
+			),
+		)
+	);
+
+	$style_engine->add_style(
+		'wp-layout-flow',
+		array(
+			'selector' => '.alignright',
+			'rules'    => array(
+				'float'        => 'right',
+				'margin-left'  => '2em',
+				'margin-right' => '0',
+			),
+		)
+	);
+
+	$style_engine->add_style(
+		'wp-layout-flow',
+		array(
+			'selector' => '> .alignfull',
+			'rules'    => array(
+				'max-width' => 'none',
+			),
+		)
+	);
+
+	$style_engine->add_style(
+		'wp-layout-flow--global-gap',
+		array(
+			'selector' => '> *',
+			'rules'    => array(
+				'margin-top'    => '0',
+				'margin-bottom' => '0',
+			),
+		)
+	);
+
+	$style_engine->add_style(
+		'wp-layout-flow--global-gap',
+		array(
+			'selector' => '> * + *',
+			'rules'    => array(
+				'margin-top'    => 'var( --wp--style--block-gap )',
+				'margin-bottom' => '0',
+			),
+		)
+	);
+}
+
+function gutenberg_generate_common_flex_layout_styles() {
```

Comment:
I could be mistaken, but I understood "common" to be the "fixed" styles required for a layout to work, so common to every container that uses a particular layout, e.g., if a bunch of Group Blocks use a flex layout, then the "common style" is `display: flex` for flex. 

Does that sound legit?

Reply:
> I understood "common" to be the "fixed" styles required for a layout to work, so common to every container that uses a particular layout, e.g., if a bunch of Group Blocks use a flex layout, then the "common style" is display: flex for flex.

Yes, that's what I was thinking of, too — common in the sense that they're styles that will be output to the global stylesheet irrespective of the blocks contained in the current post that's being rendered. (If we settle on an approach, then there'll be some a fair bit of documentation / doc comments to add!)

In that category, I was imagining we'd have things like `display: flex` and possibly each of the different alignment values, since they're a fairly small fixed set of options, but probably not content width since that could be any arbitrary value. Although, perhaps we could include the parent / global level content width if it's set, since that will likely be shared by a bunch of different blocks?

But the goal I was interested in exploring was to see how far we can limit the amount of unique styles we generate.